### ISSUE TITLE
Tune .spec files, so that customizations are possible/easy/supportable

### ISSUE THREAD

**Issue by fgeorgatos** [plabrop/bic#16]
Here is an example of what goes in production for us, during 2016q2:

```
# find /etc/site -type f -ls
797107    4 -rw-r--r--   1 root     root         3292 Apr  5 16:09 /etc/site/lmod/SitePackage.lua
797111    4 -rw-r--r--   1 root     root          350 Apr  5 18:13 /etc/site/modules/HPCBIOS/2016q1
797110    4 -rw-r--r--   1 root     root          352 Apr  5 16:09 /etc/site/modules/HPCBIOS/20160201
797109    4 -rw-r--r--   1 root     root          350 Apr  5 16:09 /etc/site/modules/HPCBIOS/2015q4
797108    4 -rw-r--r--   1 root     root          350 Apr  5 16:09 /etc/site/modules/HPCBIOS/2015q2
887525    4 -rw-r--r--   1 root     root         1695 Apr  6 16:29 /etc/site/modules/use.own.eb/prepend
887524    4 -rw-r--r--   1 root     root         1694 Apr  6 16:29 /etc/site/modules/use.own.eb/append
797113    4 -rw-r--r--   1 root     root           53 Apr  5 16:09 /etc/site/profile.d/00_root.sh
797112    4 -rw-r--r--   1 root     root           98 Apr  5 16:09 /etc/site/profile.d/00_root.csh
797117    4 -rw-r--r--   1 root     root          401 Apr  5 16:09 /etc/site/profile.d/z01_lmod_term.sh
797116    4 -rw-r--r--   1 root     root          392 Apr  5 16:09 /etc/site/profile.d/z01_lmod_term.csh
797114    4 -rw-r--r--   1 root     root         1605 Apr  7 11:03 /etc/site/profile.d/01_modules_init.sh
797286    4 -rw-r--r--   1 root     root         1958 Apr  7 11:03 /etc/site/profile.d/01_modules_init.csh
```

if we had used currently supplied .spec files, we would be missing so far the following points:
- [x] ability to use (and _restore_) `LMOD_SYSTEM_DEFAULT_MODULES` (ie. z01_lmod_*)
- [x] ability to get module logging (copy `SitePackage.lua`, let `LMOD_PACKAGE_PATH` point to it) #37
- [x] site modules (ie. where would `$MODULEPATH` get defined to value, say, /etc/site/modules ?)
  - [x] doing above in guarded variable way is important, for the known reasons around `module use`
- [x] `00_root.csh` is missing # OK: I doubt HPC people use FreeBSD, where this would matter :-P ,
  - [x] yet it could still be useful for skipping initializations while debugging environmental things
- [x] implement `$LMOD_DISABLE` #38

fyi. there has been a 6.4.2 release, which seems to have got the "it works" from several sides.

**Comment by fgeorgatos** [comment]
ok, after reporting the `module use` bug to redhat/fedora, many others fixed their issues, as of Lmod/6.3.1: 
https://bodhi.fedoraproject.org/updates/Lmod-6.3.1-3.el6

**Comment by fgeorgatos** [comment]
although module logging is not enabled per se in `Lmod/7.4.9`, it is not blocked either and has a known fix path.

so, I consider this issue closed after the releases of that RPM!
